### DISCUSSION TITLE
problem with grammar in ANTLR Lab

### DISCUSSION THREAD

**Discussion by rwblues** [antlr/antlr4#4468] [Q&A] [answered]
hallo, the grammar is:

grammar TextBlock;
textBlock: sentence+ EOF;
sentence: word+ FullStop;
word: LETTER+;
LETTER: [a-zA-Z];
FullStop: '.';

This is the input: 
This is a text.
It consists of several lines.
Each line consists of one or more Words and is terminated with a fullstop.

And here is the response:
1:14 token recognition error at: '.'
2:28 token recognition error at: '.'
3:73 token recognition error at: '.'
1:0 mismatched input 'This' expecting 'and'
3:29 extraneous input 'more' expecting {, 'and'}
3:44 extraneous input 'is' expecting {'and', '.'}

The following modification did not help: sentence: word+ FullStop {getText().length() > 1};

Please give advice. 
Thanks

**Comment by kaby76** [comment, marked as answer]
Make sure you click on the lexer source code tab and erase everything in that textbox. You have a 'combined grammar' so you only put the combined grammar in 'Parser' textbox. The lab.antlr.org website uses an interpreter of the grammar, so actions `{getText().length() > 1}` don't work. By the way that action won't work for a generated parser because the code won't compile...missing semicolon.

**Reply by kaby76** [reply]
Reproduced. It is caused exactly because you did not clear out the lexer grammar from the lexer textedit box in lab.antlr.org. The grammar still has issues, though. Suggest this:
```
grammar TextBlock;
textBlock: sentence+ EOF;
sentence: word+ FullStop;
word: LETTER+;
LETTER: [a-zA-Z];
FullStop: '.';
WS: [ \n\r\t]+ -> channel(HIDDEN);
```

**Reply by rwblues** [reply]
THANK YOU!
Now I have a basis on which I can continue. The graphical output does not show the individuel words but every single letter is reproduced. At least no errors are shown. 
Thanks again

**Comment by DavidAntliff** [comment]
Man, I only use ANTLR Lab once or twice a year but this problem _always_ trips me up - it's so non-obvious to me that there's a Lexer "tab" that needs to be cleared. I typically waste 30-60 minutes trying to get a basic grammar to work, before I notice the "Lexer" button, which doesn't even look like a button.

I hope that my comment comes up in my search engine when I encounter this again in late 2024...

(For whatever reason, the default view when I visit the site is the Parser tab, yet the warning to clear the Lexer content is on the Lexer tab... perhaps there should be a similar warning on the Parser tab?)
